common/src/test/java/io/netty/util/RecyclerTest.java

```diff
@@ -24,7 +24,7 @@
 
 public class RecyclerTest {
 
-    @Test(expected = IllegalStateException.class)
+    //@Test(expected = IllegalStateException.class)
```

Comment:
@ninja- maybe I was not clear... If you consider removing safety checks I would like to get a proof that it is actually such a big deal that it worth it. If there is no notable performance gain (which I suspect here) I don't think it makes sense to allow to remove the safety checks.